nuxeo/nuxeo.py

```diff
@@ -222,23 +259,20 @@ def execute(
             'X-NXproperties': '*',
             # Keep compatibility with old header name
             'X-NXDocumentProperties': '*',
+            'X-NXRepository': self._repository,
         }
         if void_op:
-            headers.update({'X-NXVoidOperation': 'true'})
-        headers.update({'X-NXRepository': self._repository})
+            headers['X-NXVoidOperation'] = 'true'
         if extra_headers:
             headers.update(extra_headers)
-        headers.update(self._get_common_headers())
 
         json_struct = {'params': {}}
         for k, v in params.iteritems():
             if v is None:
                 continue
             if k == 'properties':
                 if isinstance(v, dict):
-                    s = ''
-                    for propname, propvalue in v.iteritems():
-                        s += '%s=%s\n' % (propname, propvalue)
+                    s = '\n'.join(['{}={}'.format(name, value) for name, value in v.iteritems()])
```

Comment:
Could you add a test with properties containing exotic characters? I suspect this section to be fragile, and it will prevent issues with Python 3 support.
You change lines in _test_document.py_ `test_document_create_properties_as_str` and `test_document_create` as it already test for properties.